test/e2e/util.go

```diff
@@ -509,6 +511,40 @@ func waitForPodsRunningReady(ns string, minPods int, timeout time.Duration) erro
 	return nil
 }
 
+func podFromManifest(filename string) *api.Pod {
+	var pod api.Pod
+	Logf("Parsing pod from %v", filename)
+	data, err := ioutil.ReadFile(filename)
+	Expect(err).NotTo(HaveOccurred())
+
+	json, err := utilyaml.ToJSON(data)
+	Expect(err).NotTo(HaveOccurred())
+
+	Expect(runtime.DecodeInto(api.Codecs.UniversalDecoder(), json, &pod)).NotTo(HaveOccurred())
+	return &pod
+}
+
+// Run a test container to try and contact the Kubernetes api-server from a pod, wait for it
+// to flip to Ready, log its output and delete it.
+func runKubernetesServiceTestContainer(repoRoot string, ns string) {
+	c, err := loadClient()
+	if err != nil {
+		Logf("Failed to load client")
+		return
+	}
+	p := podFromManifest(filepath.Join(repoRoot, "test", "images", "clusterapi-tester", "pod.yaml"))
+	p.Namespace = ns
+	_, err = c.Pods(ns).Create(p)
+	Expect(err).NotTo(HaveOccurred())
```

Comment:
If error occurs here, will the whole e2e failed with this error? 
Or should we just log error here and continue with the "not all pods are ready" failure?

Same comment apply to all expect statement in this method...